src/main/java/org/asamk/signal/http/HttpServerHandler.java

```diff
@@ -89,7 +89,8 @@ private void handleRpcEndpoint(HttpExchange httpExchange) throws IOException {
             return;
         }
 
-        if (!"application/json".equals(httpExchange.getRequestHeaders().getFirst("Content-Type"))) {
+        if (httpExchange.getRequestHeaders().getFirst("Content-Type") == null
+                || !httpExchange.getRequestHeaders().getFirst("Content-Type").contains("application/json")) {
```

Comment:
I think using `startsWith` instead of `contains` would be more correct.
Also please extract the `httpExchange.getRequestHeaders().getFirst("Content-Type")` to a new variable, should make this a bit more readable.